### ISSUE TITLE
Provide "continue where you left off PLUS"

### ISSUE THREAD

**Issue by GitKrow** [stefanXO/Tab-Manager-Plus#5]
Dear developer:

Thank you for this tool. It's handy too use, and looks pretty too. I think it also has the potential to solved one of my problem: continue where I left off. 

As you may know, such feature IS provided by chrome. But it only records the tabs in the window that get closed the last. So if I have many tabs in many windows, I need to either migrate them in the same window, or just open the important tabs in one window in the first place. Most of the time I just don't shutdown my laptop at all, but sometimes you just have to restart. 

I can bear with what I have, but it is not ideal for me. Since this tool is already capable of managing all the tabs in all the windows, I hope you can provide a feature to create a snapshot of all the windows and tabs, and allow me to open all the tabs and windows later. It would be "continue where you left of PLUS" for me. 

After all, thank you again for making this tool available.

Best regards

**Comment by mastef** [comment]
Hey @GitKrow, so:

Scenario 1
- when I close Chrome with 3 windows open, with 10 tabs each
- then I open Chrome again and it asks me to restore the tabs
- it will open 3 windows, with 10 tabs each

Scenario 2
- when I close Chrome with 3 windows open, with 10 tabs each
- then I open Chrome again and it don't restore the tabs
- I will see the 3 windows, with 10 tabs each in the history view, and can re-open them from there. ( It will show 3 entries with "10 tabs" "10 tabs" "10 tabs" : 
![image](https://user-images.githubusercontent.com/762297/42864707-964c6052-8a91-11e8-8a14-f6e47bb46721.png)


My Chrome doesn't behave in the way as yours, that it would only re-open 1 window. Make sure you're on the latest version - or maybe I misunderstood something?

**Comment by GitKrow** [comment]
Hi Mastef,

Thanks for your reply. You understood me perfectly. I think my chrome is behaving differently from yours. And it is the latest version (I don't like it's new look BTW). In history I can see similar things. But it will only restore the one window/tab on the top of the list, doesn't matter if I closed them sequentially or did so by right clicking in the task bar and choosing close all windows. 

It could be the difference (bug?) between Mac and Windows version? As I observe the same thing across multiple Windows machines.

**Comment by mastef** [comment]
I'm just having a panicky thought - could you try with Tab Manager Plus disabled? Or set "Limit Tabs Per Window" to 0. I hope it's not the "Limit Tabs Per Window" option that's doing anything weird for you.

**Comment by GitKrow** [comment]
No no no, don't worry about that. My experience with Chrome has been like
this for years. And I have been treating it as a flaw instead of a bug all
the time. I wonder if it's just me or all users of Windows Chrome have the
same issue and they just never notice that it is an issue. Such
discrimination! JK. I am gonna try disabling some extensions and test it
over.

**Comment by YipYipX4** [comment]
@GitKrow Are you closing windows one by one?  Then, I think it is normal that only the last closed window is restored.  Try exiting Chrome using the "Exit" command on the "vertical ellipsis" menu while you have many windows open (or just restart your computer without explicitly closing any Chrome windows).

Also, I currently use the "[Simple Window Saver](https://chrome.google.com/webstore/detail/simple-window-saver/fpfmklldfnlcblofkhdeoohfppdoejdc?hl=en)" extension that lets me name Chrome windows, close windows at any time, and then restore any named window from its menu.  I can restore a window for some topic, browse a bit or drag an additional tab into the window, then close the window, and the named window will be kept up-to-date.  Chrome also has "Bookmark all tabs..." when you right-click on a tab, but this is a manual process and not kept up-to-date automatically.

I still need a better way to search and organize the tabs, though, hence I've just discovered Tab Manager Plus.

@mastef The Simple Window Saver extension I mentioned above is also [open source](https://github.com/nickbaum/Simple-Window-Saver).  It would be awesome if its functionality could be merged into Tab Manager Plus!  I'm not an expert on Chrome extensions, but I browsed the code on github and it looks like less than 1000 lines of code total (across a few files).  I know that it saves its data in localStorage because I was once able to manually copy the localStorage data from one machine to another when I upgraded my hardware (overwriting any saved windows on the new machine).
